wallet-new/src/Cardano/Wallet/Kernel.hs

```diff
@@ -20,6 +19,11 @@ module Cardano.Wallet.Kernel (
   , walletLogMessage
   , walletPassive
   , wallets
+    -- * The only effectful getter you will ever need
+  , getWalletSnapshot
+    -- * Read-only, pure getters
+  , accountUtxo
```

Comment:
What about not exporting these at all, and having a separate module with the read-only functions on the snapshot? (I think we might already have such a module, but it's possible that we don't have a "top-level" one.) That way when people add _new_ read only functions they're not tempted to make them effectful here, because they'd be adding them to the module with all the rest of them and there's not a single effect in sight.